### PR TITLE
hash-cache-tool: Adds command to diff state

### PR DESCRIPTION
**Pull request by brooksprumo**
#### Problem

The hash cache tool cannot diff the final state of the accounts hash cache directories.

#### Summary of Changes

Adds command to diff state.

### CODE REVIEW

File: accounts-db/accounts-hash-cache-tool/src/main.rs [hunk 1]
```diff
@@ -102,6 +105,29 @@ fn main() {
...
+                                .value_name("PATH1")
+                                .help("Accounts hash cache directory 1 to diff"),
+                        )
+                        .arg(
+                            Arg::with_name("path2")
+                                .index(2)
+                                .takes_value(true)
+                                .value_name("PATH2")
+                                .help("Accounts hash cache directory 2 to diff"),
+                        ),
```
Comment (HaoranYi) [inline accounts-db/accounts-hash-cache-tool/src/main.rs:130]: https://anza-xyz.slack.com/archives/C0735T9CVPU/p1722644231493859?thread_ts=1722614955.216019&cid=C0735T9CVPU

Maybe add an optional sub-arg for bin-range to extract?
default to all bin-range.

Comment (brooksprumo) [reply accounts-db/accounts-hash-cache-tool/src/main.rs:130]: Yeah that should be possible. Can that be a follow up PR? Can also add a cli arg for the number of bins.

Comment (HaoranYi) [reply accounts-db/accounts-hash-cache-tool/src/main.rs:130]: Yeah, we can do it as a follow up PR.

**Review by HaoranYi — APPROVED** [review]
lgtm.

**Pull request merged**